core/filtermaps/chain_view.go

```diff
@@ -93,6 +103,18 @@ func (cv *ChainView) getReceipts(number uint64) types.Receipts {
 	return cv.chain.GetReceiptsByHash(blockHash)
 }
 
+// SharedRange returns the block range shared by two chain views.
+func (cv *ChainView) SharedRange(cv2 *ChainView) common.Range[uint64] {
+	if cv == nil || cv2 == nil {
+		return common.Range[uint64]{}
+	}
+	var sharedLen uint64
```

Comment:
Shall we extend the canonical first?

```go
	cv.extendNonCanonical()
	cv2.extendNonCanonical()
```

I am not sure if it's possible. But the two chain views might be constructed with the different canonical chain. It's possible that these two views doesn't have any common ancestor?

By extending the chain view first, we can make sure at least the lowest one block in the view is canonical?

Reply:
You are right, and we also need to access `hashes` under the mutex now that there is one. I'll fix this one now.

Reply:
Fixed